app/controllers/application_controller.rb

```diff
@@ -1,9 +1,19 @@
 class ApplicationController < ActionController::Base
   before_action :set_am_host
+  before_action :set_page_title
 
   protect_from_forgery with: :exception
   before_filter :configure_permitted_parameters, if: :devise_controller?
 
+  rescue_from StandardError do |exception|
```

Comment:
In fact we don't need to catch routing errors - they are caught by the catch-all route at the bottom of routes.rb.  
the logstash gem will write a standard request log record, but won't have the referrer which we want, so we want all errors to go through this in order to write the specialized logstash record so that we can report on them accurately in kibana.